.buildkite/generate_pipeline.py

```diff
@@ -0,0 +1,96 @@
+import yaml
+import json
+import pathlib
+import sys
+import os
+
+CONTAINER_VERSION = "v12"
+DOCKER_PLUGIN_VERSION = "v3.8.0"
+
+X86_AGENT_TAGS = os.getenv('X86_LINUX_AGENT_TAGS')
+AARCH64_AGENT_TAGS = os.getenv('AARCH64_LINUX_AGENT_TAGS')
+DOCKER_PLUGIN_CONFIG = os.getenv('DOCKER_PLUGIN_CONFIG')
+
+PLATFORMS = ['x86_64.metal', 'arm.metal']
+OSS = ['linux']
+
+ROOT_PATH = pathlib.Path(__file__).parent.resolve()
+
+
+def check_step(step):
+    assert step.get('label'),\
+        "Label missing."
+    assert step.get('commands'),\
+        "Commands missing."
+    agents = step.get('agents')
+    assert agents,\
+        "Agents missing."
+    assert agents.get('platform') in PLATFORMS,\
```

Comment:
I remembered that the platform was not actually required, e.g. the commit format test can be run on any of the available platforms. So we could check the value only if 'platform' was provided by the user. @andreeaflorescu what do you think?